schema-lending.graphql

```diff
@@ -31,29 +32,96 @@ type Token @entity {
   " Smart contract address of the token "
   id: ID!
 
-  " Mirrored from the smart contract "
+  " Name of the token, mirrored from the smart contract "
   name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
   symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
   decimals: Int!
+
+  " Optional field to track the price of a token, mostly for caching purposes "
+  lastPriceUSD: BigDecimal
+
+  " Optional field to track the block number of the last token price "
+  lastPriceBlockNumber: BigInt
 }
 
 enum RewardTokenType {
+  " For reward tokens awarded to LPs/lenders "
   DEPOSIT
+
+  " For reward tokens awarded to borrowers "
   BORROW
 }
 
 type RewardToken @entity {
-  " Smart contract address of the reward token "
+  " { Reward token type }-{ Smart contract address of the reward token } "
   id: ID!
 
-  " Mirrored from the smart contract "
-  name: String!
-  symbol: String!
-  decimals: Int!
+  " Reference to the actual token "
+  token: Token!
 
+  " The type of the reward token "
   type: RewardTokenType!
 }
 
+enum LendingType {
+  " Collateralized Debt Position (CDP) protocols have singular isolated positions created by users. We aggregate them to give a single view of a market "
+  CDP
+
+  " Pooled protocols pool all users assets into a single market "
+  POOLED
+}
+
+enum RiskType {
+  " Global risk means each users position in a market is combined for one score to determine if they can be liquidated "
+  GLOBAL
+
+  " Isolated risk means each users position in a market or CDP is isolated for risk of liquidation "
+  ISOLATED
+}
+
+enum InterestRateType {
+  " Stable interest rate (e.g. Aave) "
+  STABLE
+
+  " Variable interest rate (e.g. Compound) "
+  VARIABLE
+
+  " Fixed term interest rate (e.g. Notional) "
+  FIXED_TERM
+}
+
+enum InterestRateSide {
+  " Interest rate accrued by lenders "
+  LENDER
+
+  " Interest rate paid by borrowers "
+  BORROWER
+}
+
+type InterestRate @entity {
+  " { Interest rate side }-{ Interest rate type }-{ Market ID }-{ Counter } "
```

Comment:
what does Counter mean?

Reply:
You can have more than one rate. I will add doc